src/plugins/unified_search/public/dataview_picker/change_dataview.tsx

```diff
@@ -111,6 +116,20 @@ export function ChangeDataView({
     fetchDataViews();
   }, [data, currentDataViewId, adHocDataViews]);
 
+  const pendingIndexMatch = useRef<undefined | NodeJS.Timeout>();
+  useEffect(() => {
+    async function checkIndices() {
+      if (dataViewSearchString !== '' && noDataViewMatches) {
+        const matches = await kibana.services.dataViewEditor.getIndices(dataViewSearchString);
```

Comment:
what about using the DataView contract here ? 
```
const adhocDataView = await kibana.services.dataViews.create({ title dataViewSearchString });
const matches = adhocDataView.matchingIndices;
```

if we want to introduce a new API I would prefer to see it on the dataViews plugin